### ISSUE TITLE
any way to get the old api back

### ISSUE THREAD

**Issue by accessyapps** [oobabooga/text-generation-webui#5353]
Hi. I am trying to use tavern.ai with the textgeneration webui. How ever there seams to be an issue with the new openai compatible api and tavern.ai. I already tried to copy over the api extension to the new webui but it does not load and errors out. Is there a way to get the old cobold api back?
p.s: i tried with openai reverse proxi, how ever apparently tavern expects a fast response on that mode and since the webui takes a while to work it disconnects after about 20 seconds.

**Comment by Ph0rk0z** [comment]
You can rollback textgen to a time it worked.

**Comment by deevis** [comment]
Would have been nice to keep the old api (cobold?) as an available option via configuration flags.  I'm debating between refactoring all the code I've written against it (to use the new openai api) or rolling back text-generation-webui.

**Comment by RuntimeRacer** [comment]
Yeah, I ran into same issue with my Project; I've built in a simple user Auth Module for rate limiting my backend (which runs textgen) and endpoints for retrieving character chat history if users want to use our software against a local textgen instance instead.
However there is no other way than adapting to the new API for this I think; since especially backend modules for using Quantization and better performance are being updated weekly, with current evolution speed of the tech.
Feel free to check out my fork if you need inspiration; porting was easier than I thought at first: https://github.com/oobabooga/text-generation-webui/compare/main...harmony-ai-solutions:text-generation-webui-harmony-ai:main

**Comment by goodglitch** [comment]
> You can rollback textgen to a time it worked. Btw, working great on sillytavern

No you can't. I decided to update oobabooga and now none of my bots is working. I have tried 2023-10-29 and 2023-11-05 snapshots but one_click.py is broken.

**Comment by Ph0rk0z** [comment]
I don't use the scripts, you can manually install the requirements.txt Probably try re-using the environment first and see what breaks then roll back as needed.
